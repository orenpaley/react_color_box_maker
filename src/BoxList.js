import Box from "./Box";
import NewBoxForm from "./NewBoxForm";
import { useState } from "react";
import { v4 as uuid } from "uuid";

const BoxList = () => {
  const INITIAL_STATE = [
    { height: 250, width: 250, backgroundColor: "red", id: uuid() },
    { height: 150, width: 350, backgroundColor: "orange", id: uuid() },
    { height: 350, width: 150, backgroundColor: "green", id: uuid() },
  ];
  const [Boxes, SetBoxes] = useState(INITIAL_STATE);
  const addBox = (newBox) => {
    SetBoxes((boxes) => [...boxes, { ...newBox, id: uuid() }]);
  };
  const removeBox = (boxToRemove) => {
    SetBoxes((Boxes) => Boxes.filter((b) => b.id !== boxToRemove.id));
  };
  return (
    <div>
      <h1>Box Maker</h1>
      <NewBoxForm addBox={addBox} />
      <div className="BoxList">
        {Boxes.map(({ width, height, backgroundColor, id }) => (
          <Box
            key={id}
            id={id}
            width={width}
            height={height}
            backgroundColor={backgroundColor}
            remove={removeBox}
          />
        ))}
      </div>
    </div>
  );
};

export default BoxList;
