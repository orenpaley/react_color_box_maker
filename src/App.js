import "./App.css";
import BoxList from "./BoxList.js";
import "./BoxList.css";

function App() {
  return (
    <div className="App">
      <BoxList />
    </div>
  );
}

export default App;
