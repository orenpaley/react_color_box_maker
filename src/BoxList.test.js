import BoxList from "./BoxList";

import { render, fireEvent, getAllByLabelText } from "@testing-library/react";
import App from "./App";

// test("renders learn react link", () => {
//   render(<BoxList />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

it("renders without crashing", () => {
  render(<BoxList />);
});
// it("matches snapshot", function () {
//   const { asFragment } = render(<BoxList />);
//   expect(asFragment()).toMatchSnapshot();
// });

// it("should add have default items", function () {
//   const { queryByText } = render(<BoxList />);
//   const boxes = queryByText("height");
//   console.log(boxes);
//   const width = getByLabelText("Width");
//   const color = getByLabelText("Color");
//   const btn = queryByText("Add Box");
//   expect(queryByText("Product Name: Chocolate Milk")).toBeInTheDocument();
//   fireEvent.change(input, { target: { value: "Chocolate Milk" } });
//   fireEvent.click(btn);
//   expect(queryByText("Product Name: Chocolate Milk")).toBeInTheDocument();
// });

// it("should add new item", function () {
//   const { queryByText, getByLabelText } = render(<BoxList />);
//   const height = getByLabelText("Height");
//   const width = getByLabelText("Width");
//   const color = getByLabelText("Color");
//   const btn = queryByText("Add Box");
//   expect(queryByText("Product Name: Chocolate Milk")).not.toBeInTheDocument();
//   fireEvent.change(input, { target: { value: "Chocolate Milk" } });
//   fireEvent.click(btn);
//   expect(queryByText("Product Name: Chocolate Milk")).toBeInTheDocument();
// });
