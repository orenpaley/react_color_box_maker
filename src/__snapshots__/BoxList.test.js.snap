// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Box Maker
    </h1>
    <form>
      <label
        for="height"
      >
        Height: 250
      </label>
      <input
        id="height"
        max="500"
        min="25"
        name="height"
        placeholder="Box Height"
        step="50"
        type="range"
        value="250"
      />
      <label
        for="width"
      >
        Width:250
      </label>
      <input
        id="width"
        max="500"
        min="25"
        name="width"
        step="50"
        type="range"
        value="250"
      />
      <label
        for="backgroundColor"
      >
        Color:green
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        type="text"
        value="green"
      />
      <button>
        Add Box
      </button>
    </form>
    <div
      class="BoxList"
    >
      <div
        id="b5f53953-5bf9-428b-9f00-dadf47d3d0e2"
        style="width: 250px; height: 250px; border: 2px solid black; background-color: red;"
      >
        <button>
          X
        </button>
      </div>
      <div
        id="788f5553-2fda-4d36-94c8-19f25a0bc519"
        style="width: 350px; height: 150px; border: 2px solid black; background-color: orange;"
      >
        <button>
          X
        </button>
      </div>
      <div
        id="9f41cf58-88e9-4d10-8aa9-a3bae5317a06"
        style="width: 150px; height: 350px; border: 2px solid black; background-color: green;"
      >
        <button>
          X
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
